go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -344,17 +328,6 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 		return nil
 	}
 
-	if bsnc.isPodScheduledinLocalZone(pod) {
-		return bsnc.removeLocalZonePodForSecondaryNetwork(pod, portInfoMap)
-	}
-
-	// For remote pods, we just need to remove the pod IPs from the pod namespace address set
-	return bsnc.removeRemoteZonePodFromNamespaceAddressSet(pod)
```

Comment:
where does this happen for non layer2 networks now?

Reply:
Yeah, that was missing. Added corresponding UTs.

For some reason, that made me think we need to be aware of nodes zone on localnet as well. For some other reason, I did not realize this before. It just works anyway, as the LSPs we create for remote pods are never bound and they don't really affect functionally. This being the case, I will rework it on a follow-up PR.